src/jekyll/jk_document.rs

```diff
@@ -0,0 +1,219 @@
+use liquid;
+
+use super::super::frontmatter;
+use super::super::datetime;
+use std::fs::{File, create_dir_all};
+use std::io::Write;
+use std::path::Path;
+use super::super::document::{read_file, split_document};
+use frontmatter::FrontmatterBuilder;
+use error::{ErrorKind, Result};
+use serde_yaml;
+
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
+#[derive(Serialize, Deserialize)]
+pub struct JkFrontmatterBuilder(liquid::Object);
+
+impl JkFrontmatterBuilder {
+    pub fn new() -> JkFrontmatterBuilder {
+        JkFrontmatterBuilder(liquid::Object::new())
+    }
+}
+
+impl From<JkFrontmatterBuilder> for frontmatter::FrontmatterBuilder {
+    fn from(jk_front: JkFrontmatterBuilder) -> Self {
+        // Convert jekyll frontmatter into frontmatter (with `custom`)
+        let mut custom_attributes = jk_front.0;
+        frontmatter::FrontmatterBuilder::new()
+            .merge_title(custom_attributes.remove("title").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_slug(custom_attributes.remove("permalink").and_then(|v| {
+                v.as_str().map(|s| s.to_owned())
+            }))
+            .merge_description(custom_attributes.remove("excerpt").and_then(|v| {
```

Comment:
Jekyll doesn't have a description?  I thought I came across it when digging in the code

Reply:
I couldn't find it, maybe I oversight it, will triple check.

Reply:
Didn't cross any `description` field in Jekyll's documentation.  On the other hand, I don't see an `excerpt` field in cobalt, so I made this mapping

Reply:
Oh well.  Can always be modified later if we find out more.